### PR TITLE
Update Data Model for Mission Sorting and Form 791

### PR DESCRIPTION
**Pull request by jbethanco**
[PBS-1451](https://jira.il2.dso.mil/browse/PBS-1451)

Add Form791 Entity for Tankers
Add form791 property to Sortie accessors file
Add Issue Entity to support 791 fuel issues
Remove combat time and combat support time from FlightConditions
(combat/combat support time is now calculated based on mission type)

Remove all mentions and usages of combat and combat support time from
auditors and FlightConditions+Encodable

Add necessary supporting functions to AllEntityAwake in support of migration to new data model
Add expiring TODO comments to AllEntityAwake so we can remove unnecessary migration code in a few months

Change Mission Sort to new sortingDate
Change sortie takeoffTime to takeoffTime_ to enable new setter / getter
Create new setter for Sortie takeoffTime that updates the mission sortingDate

Add fillable DD Form 791 Form to Resources Folder in prep for output

Fix multiple unit test's dataController loading order, to prevent test crash
Update SceneDelegate init so that UITests use an in memory core data store
Correct Sortie.metrics deletion method to Cascade from Delete

Regenerate all core data entities class and property files

✅ PLEASE TEST Data Migration in simulator and on device:
1. Run main branch.
2. Edit missions, training events, sortie types , etc
3. Update to this branch
4. View/Edit missions, training events, sortie types, etc

Pay close attention to takeoff and land times. 
If you can make a crash, please show error and steps to repeat. 
If you can find data that does not carry over, please describe.

### CODE REVIEW

**Review by mcgarrys — COMMENTED** [review]
Code looks super clean 🔥 🧼
Only strange behavior I noticed was with the sorting in the new branch 🤔

- I tapped on an older mission and entered the mission data view for one of it's sorties, and then returned the main missions list, where that older mission was now on the bottom of the list. 
- I tried doing the same with a newer mission, and it had the opposite effect, sending that mission to the top of the list.
- Terminating/Re-launching the app then resets the sorting correctly

Another strange sorting behavior:
- I also noticed after editing takeOff times in several different missions, the sorting stops working (assuming that missions are sorted by their oldest sortie). A mission with an earliest takeOff time on Nov 1st appears above a mission with an earliest takeOff time on Oct 31st, and this persists after terminating the app.

For testing the migration, I did the following:
- Ran on main
- Edited mission title/summary, every entity in mission data for a sortie
- Attempted to run on new branch, and ran into this: 
Screen Shot 2021-07-13 at 12 36 58 PM
- I then switched back to main, ran successfully
- Switched to the new branch again and this time ran successfully.
- I then edited every field again, now on the new branch, and it saved with no issues after closing/reopening the app.

I'm thinking this was a one-off error because I couldn't reproduce after switching from main to the new branch again (although it is interesting the error appears to have to do with combatSupportTime/combatTime). 

All data appeared to carry over, including takeOff and landingTime 🤷‍♂️ ✅

File: PBLogging/ViewModels/Auditing/FlightConditionsRow.swift [hunk 1]
```diff
@@ -12,15 +12,13 @@ struct FlightConditionRow {
     var flightAuthDutyCode: FlightAuthDutyCode
     var crewLine: CrewLine?
     var timeToDistribute = 0.0
-    var night, instruments, simInstruments, nvg, combatTime, combatSupportTime: DoubleItem
```
[the remark this answers is not in the document]
Comment (jbethanco) [reply PBLogging/ViewModels/Auditing/FlightConditionsRow.swift:15]: Nice catch.

File: PBLogging/Data/EntityExtensions/Issue+Comparable.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//
+
+import Foundation
+
+extension Issue: Comparable {
+
+    public static func < (lhs: Issue, rhs: Issue) -> Bool {
+        guard let lhsDate = lhs.dateCreated else { return true }
+        guard let rhsDate = rhs.dateCreated else { return false }
+        return lhsDate > rhsDate
```
[the remark this answers is not in the document]
Comment (jbethanco) [reply PBLogging/Data/EntityExtensions/Issue+Comparable.swift:15]: I think I'm going to leave it alone for now. Not sure if the booms want to see it with most recent on top, like sorties and missions, or if they want to see it most recent on bottom like the actual form.

**Comment by jbethanco** [comment]
> Code looks super clean 🔥 🧼
> Only strange behavior I noticed was with the sorting in the new branch 🤔
> 
> * I tapped on an older mission and entered the mission data view for one of it's sorties, and then returned the main missions list, where that older mission was now on the bottom of the list.
> * I tried doing the same with a newer mission, and it had the opposite effect, sending that mission to the top of the list.
> * Terminating/Re-launching the app then resets the sorting correctly
> 
> Another strange sorting behavior:
> 
> * I also noticed after editing takeOff times in several different missions, the sorting stops working (assuming that missions are sorted by their oldest sortie). A mission with an earliest takeOff time on Nov 1st appears above a mission with an earliest takeOff time on Oct 31st, and this persists after terminating the app.
> 
> For testing the migration, I did the following:
> 
> * Ran on main
> * Edited mission title/summary, every entity in mission data for a sortie
> * Attempted to run on new branch, and ran into this:
>   Screen Shot 2021-07-13 at 12 36 58 PM
> * I then switched back to main, ran successfully
> * Switched to the new branch again and this time ran successfully.
> * I then edited every field again, now on the new branch, and it saved with no issues after closing/reopening the app.
> 
> I'm thinking this was a one-off error because I couldn't reproduce after switching from main to the new branch again (although it is interesting the error appears to have to do with combatSupportTime/combatTime).
> 
> All data appeared to carry over, including takeOff and landingTime 🤷‍♂️ ✅

Confusion may be because the first takeoff time of the first sortie of a mission should decide the sort order. 
Only changing a time/date of a takeoff to be earlier than a current one should work. I might not be checking if the date is now older than another one. 
I'll look at that and retest the sorting orders here and see what I missed. Sounds like it isn't acting consistent. 

I believe xcode build errors may have been due to a dirty build folder before running the new branch. May have had some stale compiled things in there.

**Comment by jbethanco** [comment]
I believe all comments have been addressed. Seems no one was able to get a migration crash, so that is comforting. Added the missing Form791 issues property that Beth caught. 

Expected sort behavior based on feedback received by Rachael, as follows:
Missions should sort with most recent mission on top. A misson's recency is determined by its earliest sortie take off date/time. 

Sorties sort with the most recent sortie on top as determined by its takeoff time.

**Comment by bmorris-revacomm** [comment]
> I believe all comments have been addressed. Seems no one was able to get a migration crash, so that is comforting. Added the missing Form791 issues property that Beth caught.
> 
> Expected sort behavior based on feedback received by Rachael, as follows:
> Missions should sort with most recent mission on top. A misson's recency is determined by its earliest sortie take off date/time.
> 
> Sorties sort with the most recent sortie on top as determined by its takeoff time.

It's still not sorting for me 🤔

**Review by bmorris-revacomm — APPROVED** [review]
jk, looks good 🥳

**Pull request merged**